include/utility/json_utility.hpp

```diff
@@ -650,5 +650,141 @@ inline bool isBackupAndRestoreRequired(const nlohmann::json& i_sysCfgJsonObj)
+ */
+inline bool isPollingRequired(const nlohmann::json& i_sysCfgJsonObj,
+                              const std::string i_vpdFilePath)
+{
+    if (i_sysCfgJsonObj.empty() || i_vpdFilePath.empty())
+    {
+        logging::logMessage("Invalid parameters");
+        return false;
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        logging::logMessage("Missing frus section in system config JSON");
+        return false;
+    }
+
+    if (i_sysCfgJsonObj["frus"][i_vpdFilePath].at(0).contains(
+            "pollingRequired"))
+    {
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ * @brief API to return parameters required for polling.
+ *
+ * This API checks for "gpioPresence" tag under "pollingRequired" in system
+ * config JSON. If present returns the parameters for GPIO polling.
+ *
+ * @throw JSON Exception, if JSON does not contain "pollingRequired" tag, "pin"
+ * and "value" tags under the given VPD file path. And std::runtime_error for
+ * invalid parameters.
+ *
+ * @param[in] i_sysCfgJsonObj -  System config JSON object.
+ * @param[in] i_vpdFilePath - VPD file path.
+ *
+ * @return GPIO parameters if present, otherwise throws std::runtime_error.
+ */
+inline types::GpioPollingParameters
+    getGpioPollingParameters(const nlohmann::json& i_sysCfgJsonObj,
+                             const std::string i_vpdFilePath)
+{
+    types::GpioPollingParameters l_pollingParameters;
+
+    if (i_sysCfgJsonObj.empty() || i_vpdFilePath.empty())
+    {
+        throw std::runtime_error("Invalid parameters");
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in system config JSON");
+    }
+
+    if (i_sysCfgJsonObj["frus"][i_vpdFilePath]
+            .at(0)["pollingRequired"]
+            .contains("gpioPresence"))
+    {
+        std::get<0>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(
+                0)["pollingRequired"]["gpioPresence"]["pin"];
+        std::get<1>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(
+                0)["pollingRequired"]["gpioPresence"]["value"];
+        std::get<2>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(0)["inventoryPath"];
+    }
+    else
+    {
+        throw std::runtime_error(
+            "Missing gpioPresence tag under pollingRequired");
+    }
+
+    return l_pollingParameters;
+}
+
+/**
+ * @brief An API returns list of FRUs parameters that needs Polling.
+ *
+ * An API that checks for the FRUs that requires GPIO polling and returns
+ * a list of FRUs with polling parameters. Returns an empty list if there are
+ * no FRUs that requires polling.
+ *
+ * @param[in] i_sysCfgJsonObj - System config JSON object.
+ *
+ * @return list of FRUs parameters that needs polling.
+ */
+inline std::vector<types::GpioPollingParameters>
+    getPollingRequiredFrusDetails(const nlohmann::json& i_sysCfgJsonObj)
```

Comment:
Can't figure out from the name what function does.
Some more thoughts required.

Reply:
As discussed inthe broader team, lets go ahead with "getListOfPollingParamsForFrus".

Reply:
Updated